server/plugin.go

```diff
@@ -599,6 +611,30 @@ func (p *Plugin) HasGroupHook(ctx context.Context, user *gitlab.UserInfo, namesp
 	return found, err
 }
 
+// getUsername returns the GitLab username for a given Mattermost user,
+// if the user is connected to GitLab via this plugin.
+// Otherwise it returns the Mattermost username.
+func (p *Plugin) getUsername(userID string) (string, *APIErrorResponse) {
+	info, err := p.getGitlabUserInfoByMattermostID(userID)
+	if err != nil {
```

Comment:
minor: I think we can make this a bit more readable and it's ok to have some repetition.
```go
if  err != nil && if err.ID != APIErrorIDNotConnected {
      return "", err
}

if err != nil {
 // rest of logic
}